doc/Building.md

```diff
@@ -0,0 +1,199 @@
+
+If you only want to work with Lua frontend stuff, you can grab the AppImage and
+run it with `--appimage-extract`.
+
+You can skip most of the following instructions if desired, and use our premade Docker image instead. In that case the only requirements are Git and Docker. See [the virtual development environment README](https://github.com/koreader/virdevenv) for more information.
+
+## Prerequisites
+
+To get and compile the source you must have `patch`, `wget`, `unzip`, `git`,
+`cmake` and `luarocks` installed, as well as a version of `autoconf`
+greater than 2.64. You also need `nasm` and of course a compiler like `gcc`
+or `clang`. If you want to cross-compile for other architectures, you need a proper
+cross-compile toolchain. Your GCC should be at least version 4.8.
+
+### Debian/Ubuntu and derivates
+
+Install the prerequisites using dpkg:
+
+```
+sudo apt-get install build-essential git patch wget unzip \
+gettext autoconf automake cmake libtool nasm luarocks libsdl2-dev \
+libssl-dev libffi-dev libsdl2-dev libc6-dev-i386 xutils-dev linux-libc-dev:i386 zlib1g:i386
+```
+
+
+### Fedora/Red Hat
+
+Install `libstdc++-static`, `SDL` and `SDL-devel` packages using `yum`
+
+### MacOS
+
+Install the prerequisites using [Homebrew](https://brew.sh/):
+
+```
+brew install nasm binutils libtool autoconf automake cmake makedepend \
+sdl2 lua@5.1 luarocks gettext pkg-config wget md5sha1sum
+echo 'export PATH="/usr/local/opt/gettext/bin:$PATH"' >> "$HOME"/.bash_profile
+```
+
+If you run into a gettext error while building glib, try `brew link --force gettext` to override the built-in Mac OS BSD gettext with GNU GetText.
+
+*Note:* in Mojave (10.14) you need to set a minimum deployment version higher than 10.04. Otherwise you'll get the error `ld: library not found for -lgcc_s.10.4`.
+```
+export MACOSX_DEPLOYMENT_TARGET=10.09
+```
+
+
+## Getting the source
+
+
+```
+git clone https://github.com/koreader/koreader.git
+cd koreader && ./kodev fetch-thirdparty
+```
+
+Building the emulator
+
+## Building and running the emulator
+
+To build an emulator on your current Linux or MacOS machine:
+
+```
+./kodev build
+```
+
+To run KOReader on your development machine:
+
+```
+./kodev run
+```
+
+
+You can specify the size and DPI of the emulator's screen using
+`-w=X` (width), `-h=X` (height), and `-d=X` (DPI).
+
+ There is also a convenience
+`-s` (simulate) flag with some presets like `kobo-aura-one`, `kindle3`, and
+`hidpi`. The latter is a fictional device with `--screen_width=1500`,
+`--screen_height=2000` and `--screen_dpi=600` to help ensure DPI scaling works correctly.
+
+Sample usage:
+
+```
+./kodev run -s=kobo-aura-one
+```
+
+To use your own koreader-base repo instead of the default one change the `KOR_BASE`
+environment variable:
+
+```
+make KOR_BASE=../koreader-base
+```
+
+This will be handy if you are developing `koreader-base` and you want to test your
```

Comment:
Not sure I disagree with this; just using branches in your base submodule works perfectly fine. But it's potentially slightly more convenient to quickly switch back & forth between different experiments, particularly if they involve Clang vs GCC or something drastic like that.

Reply:
I never used that, but as you can imagine I copied blindly almost all the information and only tried to digest the doc for final users.